processing/src/main/java/io/druid/granularity/QueryGranularity.java

```diff
@@ -86,4 +89,74 @@ else if(o instanceof Number)
     }
     throw new IAE("Cannot convert [%s] to QueryGranularity", o.getClass());
   }
+
+  // best effort comparing
+  public static Integer compare(QueryGranularity g1, QueryGranularity g2) {
```

Comment:
@himanshug Thought on that once. But I wanted make it clear that it's just best effort try. Anyway, I'll do that.